src/Application.cpp

```diff
@@ -630,26 +631,47 @@ bool Application::loadGame(const std::string& path)
       return false;
     }
   }
+
+  _system = getSystem(_emulator, path, &_core);
     
   if (!_core.loadGame(path.c_str(), data, size))
   {
-    free(data);
+    if (_system == System::kNintendo)
+    {
+      // Assume that the FDS system is missing
+      _logger.debug(TAG "Game load failure (Nintendo)");
+
+      MessageBox(g_mainWindow, "Game load error. Are you missing a system file?", "Core Error", MB_OK);
```

Comment:
recommend showing a generic message for all systems. it's useful feedback, and there may be other reasons this fails for NES.

Reply:
That seems fine, but I want to keep this message, since a generic error message is no more helpful than crashing when a file is missing. I don't know if it is possible to differentiate between different types of errors in libretro, so the alternative would be to detect the system file ourselves before falling back to a generic error message. In any case, errors during core operations should be added at all levels, not just the loading stage, so I think another PR with better error handling would be more adequate.